Calculate.Tests/ProgramTests.cs

```diff
@@ -0,0 +1,261 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using Calculate;
+using System.Security;
+
+
+namespace Calculate.Tests;
+
+[TestClass]
+public class ProgramTests
+{
+    private static readonly string?[] Quit = new[] { "q" };
+    private static readonly string?[] QuitUpper = new[] { "Q" };
+    private static readonly string?[] EofOnly = new string?[] { null };
+    private static readonly string?[] EmptyThenQuit = new string?[] { "", "q" };
+
+    private static readonly string?[] AddThenQuit = new[] { "3 + 4", "q" };
+    private static readonly string?[] TwoOpsThenQuit = new[] { "3 + 4", "3 - 1", "q" };
+
+    private static readonly string?[] NoSpacesThenQuit = new[] { "3+4", "q" };
+    private static readonly string?[] UnknownOpThenQuit = new[] { "3 ^ 2", "q" };
+    private static readonly string?[] DivZeroThenQuit = new[] { "1 / 0", "q" };
+
+    [TestMethod]
+    public void Properties_CanBeSetAtConstructionInvokedCorrectly_Success()
+    {
+        // Arrange
+        var writes = new List<string?>();
+        var inputs = new Queue<string?>(new[] { "hello", (string?)null });
+
+        Action<string?> captureWrite = s => writes.Add(s);
+        Func<string?> supplyRead = () => inputs.Count > 0 ? inputs.Dequeue() : null;
+
+        // Act
+        var program = new Program(captureWrite, supplyRead);
+
+        program.WriteLine("first");
+        var read1 = program.ReadLine();
+        var read2 = program.ReadLine(); //will return null
+
+        // Assert
+        Assert.AreEqual<int>(1, writes.Count);
```

Comment:
nit: Unit tests should typically have a single assertion in order to keep them focused and so you know it fails for a specific reason.